src/Datadog.Trace/Sampling/RateLimiter.cs

```diff
@@ -143,8 +137,13 @@ private void WaitForRefresh()
             }
             finally
             {
-                // Resume threads
-                _refreshEvent.Set();
+                if (refreshing == 0)
+                {
+                    // If refreshing is 0, it means that this thread acquired the lock
+                    // Releasing it before leaving
+                    // Note: a full fence might not be needed here, but better safe than sorry
+                    Interlocked.Exchange(ref _refreshing, 0);
```

Comment:
When I first read this I though we were setting the value to zero if it is already zero, which made no sense.  Then I realized `refreshing` and `_refreshing` are two different things. Can we rename them to something a little more clear? Maybe the local variable can be something like `wasRefreshing` or `lockAquired`?

Reply:
Does `refreshInProgress` work?